### PR TITLE
#40 Added support for selection on SOURCE_REFERENCE column

### LINKED ISSUE #40: Support selection on SOURCE_REFERENCE column

### PR DESCRIPTION
**Pull request by jakobbraun**
Closes #40

### CODE REVIEW

File: src/main/java/com/exasol/adapter/document/QueryPlanner.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package com.exasol.adapter.document;
+
+import com.exasol.adapter.document.documentfetcher.DocumentFetcher;
+import com.exasol.adapter.document.queryplanning.RemoteTableQuery;
+
+/**
+ * This class plans the query execution and returns it as {@link QueryPlan}.
+ * <p>
+ * For that it builds one ore more {@link DataLoader}s that fetch the required documents for a given
```
Comment (morazow) [inline src/main/java/com/exasol/adapter/document/QueryPlanner.java:9]:
```suggestion
 * For that it builds one or more {@link DataLoader}s that fetch the required documents for a given
```

File: src/test/java/com/exasol/adapter/document/UdfCallBuilderTest.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        final QueryPlan queryPlan = new QueryPlan(List.of(), new NoPredicate());
+        final String udfCallSql = UDF_CALL_BUILDER.getUdfCallSql(queryPlan, remoteTableQuery);
+        assertThat(udfCallSql, equalTo(
+                "SELECT  FROM (SELECT \"ADAPTERS\".IMPORT_FROM_TEST_ADAPTER(\"DATA_LOADER\", \"REMOTE_TABLE_QUERY\", \"CONNECTION_NAME\") EMITS  FROM (VALUES ) AS \"T\"(\"DATA_LOADER\", \"REMOTE_TABLE_QUERY\", \"CONNECTION_NAME\", \"FRAGMENT_ID\") GROUP BY \"FRAGMENT_ID\") WHERE TRUE"));
+    }
+
+    @Test
+    void testAddPostSelection() throws IOException {
+        final RemoteTableQuery remoteTableQuery = getRemoteTableQueryForNoColumns();
+        final ColumnLiteralComparisonPredicate postSelection = new ColumnLiteralComparisonPredicate(
```
Comment (morazow) [inline src/test/java/com/exasol/adapter/document/UdfCallBuilderTest.java:39]: Does the post selection on source reference columns should be enabled by users or is it automatically done with where clause from users' queries?

**Review by morazow — APPROVED** [review]
Single typo, then looks good.

File: src/test/java/com/exasol/adapter/document/UdfCallBuilderTest.java (hunk 2 above)
Comment (jakobbraun) [reply src/test/java/com/exasol/adapter/document/UdfCallBuilderTest.java:39]: automatically.

**Pull request merged**